cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/resource/AzureVolumeResourceBuilder.java

```diff
@@ -220,17 +221,21 @@ public CloudResource delete(AzureContext context, AuthenticatedContext auth, Clo
 
         for (CloudResourceStatus cloudResourceStatus : removableDisks) {
             CloudResource cloudResource = cloudResourceStatus.getCloudResource();
-            String instanceName = cloudResource.getInstanceId();
             List<String> volumeIds = getVolumeSetAttributes(cloudResource).getVolumes().stream()
                     .map(VolumeSetAttributes.Volume::getId)
                     .collect(Collectors.toList());
             for (String volumeId : volumeIds) {
                 if (ResourceStatus.ATTACHED.equals(cloudResourceStatus.getStatus())) {
+                    String instanceName = cloudResource.getInstanceId();
```

Comment:
This method is too complex. Please create a follow-up task to refactor.

Reply:
Thx, pls. re-check